backend/src/validation/mod.rs

```diff
@@ -327,6 +335,118 @@ impl EventValidator {
     }
 }
 
+/// Check if an IP address is blocked by the provided blacklist entries.
+/// Supports IPv4, IPv6, and CIDR ranges (both v4 and v6).
+pub fn check_blacklist(
+    ip_address: &str,
+    blacklisted_ips: &[String],
+) -> Result<(), ValidationError> {
+    let event_ip: IpAddr = IpAddr::from_str(ip_address)
+        .map_err(|_| ValidationError::InvalidIpAddress("Invalid IP address format".to_string()))?;
+
+    let is_blocked = blacklisted_ips.iter().any(|entry| {
+        if let Ok(ip) = IpAddr::from_str(entry) {
+            ip == event_ip
+        } else if let Ok(net) = entry.parse::<IpNet>() {
+            net.contains(&event_ip)
+        } else {
+            false
+        }
+    });
+
+    if is_blocked {
+        Err(ValidationError::BlacklistedIp("IP is blacklisted".to_string()))
+    } else {
+        Ok(())
+    }
+}
+
+/// Outcome for site-config validation to support tagging in storage
+#[derive(Debug, Clone, Copy)]
+pub enum SiteConfigStatus {
+    Approved,
+    Missing,
+    FetchError,
+}
+
+impl SiteConfigStatus {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            SiteConfigStatus::Approved => "approved",
+            SiteConfigStatus::Missing => "missing",
+            SiteConfigStatus::FetchError => "fetch_error",
+        }
+    }
+}
+
+/// Provides a mapping to the ClickHouse Enum8 values
+impl From<SiteConfigStatus> for i8 {
+    fn from(value: SiteConfigStatus) -> Self {
+        match value {
+            SiteConfigStatus::Approved => 1,
+            SiteConfigStatus::Missing => 2,
+            SiteConfigStatus::FetchError => 3,
+        }
+    }
+}
```

Comment:
This mapping should be moved to `backend\src\db\models.rs`, where we're doing the rest of the enum mappings